src/middleware/csrf.ts

```diff
@@ -21,7 +21,12 @@ export default function(): Middleware {
     if (!ctx.session.user) return next();
 
     if (!safeMethods.includes(ctx.method) && !safePaths.includes(ctx.path)) {
-      ctx.validateCsrf();
+      if(ctx.path === '/change-password'){
+        ctx.validateCsrf(await ctx.getCsrf());
```

Comment:
This change unfortunately doesn't really make sense. `getCsrf()` returns the current CSRF token, so this will always be valid.

The point of `validateCsrf` is to validate a token that was submitted with the request. The solution here would be to ensure that the HTML form has an `<input type="hidden" name="csrf-token" value"..." />` included.

Reply:
Gotcha. B/c the current form doesn't have the csrf-token value, I'll try this:

1. Send csrf-token from the `/change-password` mw.
2. Attach csrf-token in `changepassword.hbs`.
3. Validate the token back in the server.

Reply:
Fixed

Reply:
Excellent!